app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetController.kt

```diff
@@ -178,9 +178,16 @@ class BloodPressureEntrySheetController @Inject constructor(
   }
 
   private fun prefillDate(events: Observable<UiEvent>): Observable<UiChange> {
-    return events
+    val localDates = events
         .ofType<DateToPrefillCalculated>()
         .map { it.date }
+
+    val updates = events
+        .ofType<SheetCreated>()
+        .filter { it.openAs is Update }
+
+    return Observables
```

Comment:
Should we use `updates.withLatestFrom(localDates)` instead?

Reply:
That was my initial approach, but one of the streams were terminated and `withLatestFrom` was never executed. Hence the `combineLatest` with a `take(1)`. Tried your code again just to be sure but had the same effect.